base_layer/chat_ffi/src/conversationalists.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright 2023, The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use std::{convert::TryFrom, ptr};
+
+use libc::{c_int, c_uint};
+use tari_chat_client::ChatClient as ChatClientTrait;
+use tari_common_types::tari_address::TariAddress;
+
+use crate::{
+    error::{InterfaceError, LibChatError},
+    ChatClient,
+};
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct ConversationalistsVector(pub Vec<TariAddress>);
+
+/// Return a ptr to a ConversationalistsVector
+///
+/// ## Arguments
+/// `client` - The ChatClient
+/// `error_out` - Pointer to an int which will be modified
+///
+/// ## Returns
+/// `*mut ptr ConversationalistsVector` - a pointer to a ConversationalistsVector
+///
+/// ## Safety
+/// The `ConversationalistsVector` should be destroyed after use
+#[no_mangle]
+pub unsafe extern "C" fn get_conversationalists(
```

Comment:
The destroy is used in the `test_retrieving_conversationalists_from_vector`.

As for a `get_conversationlists` test it gets harder because it requires a whole running client/db/service etc. Which I haven't setup for any of the unit testing in chat. It is covered in cucumber though.